src/macad_gym/carla/multi_env.py

```diff
@@ -368,25 +372,26 @@ def __init__(self, configs=None):
         self._last_obs = None
         self._image = None
         self._surface = None
-        self._obs_dict = {}
         self._video = False
+        self._obs_dict = {}
+        self._done_dict = {}
         self._previous_actions = {}
         self._previous_rewards = {}
         self._last_reward = {}
+        self._npc_vehicles = []  # List of NPC vehicles
+        self._npc_pedestrians = []  # List of NPC pedestrians
         self._agents = {}  # Dictionary of macad_agents with agent_id as key
         self._actors = {}  # Dictionary of actors with actor_id as key
-        self._path_trackers = {}
-        self._collisions = {}
-        self._lane_invasions = {}
-        self._scenario_map = {}
-        self._load_scenario(self._scenario_config)
-        self._done_dict = {}
-        self._dones = set()  # Set of all done actor IDs
+        self._cameras = {}  # Dictionary of sensors with actor_id as key
+        self._path_trackers = {}  # Dictionary of sensors with actor_id as key
+        self._collisions = {}  # Dictionary of sensors with actor_id as key
+        self._lane_invasions = {}  # Dictionary of sensors with actor_id as key
+        self._scenario_map = {}  # Dictionary with current scenario map config
 
     @staticmethod
-    def _get_free_tcp_port():
+    def _get_tcp_port(port=0):
         s = socket.socket()
-        s.bind(("", 0))  # Request the sys to provide a free port dynamically
+        s.bind(("", port))  # Request the sys to provide a free port dynamically
```

Comment:
This method is used to get a resource collision-free TCP socket that is guaranteed to be free/available. This is not meant to be a method to return a socket object given a particular port number.
Is there a particular reason you wanted to make this change to make the `port` number an argument? If not, please revert this change.

Reply:
To make it user configurable with abilty to connect from the extern? I personally launch the execution on a cluster and use the port mapped to the extern to connect. I f you don't like it, i've no problem in making a rollback.

Reply:
I see. Remote connections to the CARLA server launched through MACAD-Gym sounds like an unintended use case but if that's useful, it's okay to keep your changes since the default argument value of `0` has the same behaviour as before. Please update the comment to make it clear that only `port=0` will request the sys to provide a free port dynamically. Not any other value. 
Keep in mind that if the `port` argument value supplied to this method is already in use by some other process, this process will crash because that exception is not handled (using a try except statement).

Reply:
The `port` is not currently exposed to the user for configuration. Even if you add this argument, how will a user change it?

Reply:
i extendend the class that again, i think is not the intended use you wanted for the library 😅. I added the docstring for the method in the last commit